tools/generate-helpers/pg-table-bindings/list.go

```diff
@@ -49,10 +51,32 @@ func storageToResource(t string) string {
 
 func isGloballyScoped(storageType string) bool {
 	resource := storageToResource(storageType)
+	metadata := resourceMetadataFromString(resource)
+	return metadata.GetScope() == permissions.GlobalScope
+}
+
+func isDirectlyScoped(schema *walker.Schema) bool {
+	resource := storageToResource(schema.Type)
+	scope := resourceMetadataFromString(resource).Scope
+	clusterIDExist := false
+	namespaceExist := false
+	for _, f := range schema.Fields {
+		if strings.Contains(f.Search.FieldName, "Cluster ID") {
+			clusterIDExist = true
+		}
+		if strings.Contains(f.Search.FieldName, "Namespace") {
+			namespaceExist = true
+		}
+	}
+	return clusterIDExist && (scope == permissions.ClusterScope || (scope == permissions.NamespaceScope && namespaceExist))
```

Comment:
I find the boolean expression hard to read. How about breaking things down per scope ?
```
switch scope {
case permissions.NamespaceScope:
    return clusterIDExists && namespaceExists
case permissions.ClusterScope:
    return clusterIDExists
default:
    return true
}
```